packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-expandable-answer.tsx

```diff
@@ -64,28 +65,52 @@ export class AtomicSmartSnippetExpandableAnswer {
     this.smartSnippet = buildSmartSnippet(this.bindings.engine);
   }
 
-  public answerSizeUpdated(event: CustomEvent<{height: number}>) {
-    const {height} = event.detail;
-    this.host.style.setProperty('--full-height', `${height}px`);
-    this.showButton = height > this.maximumHeight;
+  @Watch('fullHeight')
+  public fullHeightUpdated() {
+    this.host.style.setProperty('--full-height', `${this.fullHeight}px`);
     this.host.style.setProperty(
       '--collapsed-size',
-      `${this.showButton ? this.collapsedHeight : height}px`
+      `${this.showButton ? this.collapsedHeight : this.fullHeight}px`
     );
   }
 
+  private establishInitialHeight() {
+    return new Promise<number>((resolve) => {
+      const answerElement = document.createElement(
+        'atomic-smart-snippet-answer'
+      );
+      answerElement.htmlContent = this.smartSnippetState.answer;
+      answerElement.innerStyle = this.snippetStyle;
+      answerElement.style.visibility = 'hidden';
+      answerElement.style.position = 'absolute';
+      listenOnce(answerElement, 'answerSizeUpdated', (event) => {
+        answerElement.remove();
+        resolve((event as CustomEvent<{height: number}>).detail.height);
+      });
+      this.host.parentElement!.appendChild(answerElement);
```

Comment:
By appending the temporary `answerElement` to the parent of the host, we ensure that it inherits the same part styling, which affects the height.